src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionEndpointIT.java

```diff
@@ -0,0 +1,309 @@
+import com.mashape.unirest.http.Unirest;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import javax.xml.bind.JAXBContext;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.embedded.LocalServerPort;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.rules.SpringClassRule;
+import org.springframework.test.context.junit4.rules.SpringMethodRule;
+import uk.gov.ons.ctp.common.utility.Mapzer;
+import uk.gov.ons.ctp.response.action.config.AppConfig;
+import uk.gov.ons.ctp.response.action.message.instruction.ActionAddress;
+import uk.gov.ons.ctp.response.action.message.instruction.ActionInstruction;
+import uk.gov.ons.ctp.response.action.representation.ActionPlanDTO;
+import uk.gov.ons.ctp.response.action.representation.ActionPlanPostRequestDTO;
+import uk.gov.ons.ctp.response.action.representation.ActionPostRequestDTO;
+import uk.gov.ons.ctp.response.casesvc.message.notification.CaseNotification;
+import uk.gov.ons.ctp.response.casesvc.message.notification.NotificationType;
+import uk.gov.ons.ctp.response.casesvc.representation.CaseDetailsDTO;
+import uk.gov.ons.ctp.response.sample.representation.SampleAttributesDTO;
+import uk.gov.ons.tools.rabbit.Rabbitmq;
+import uk.gov.ons.tools.rabbit.SimpleMessageBase;
+import uk.gov.ons.tools.rabbit.SimpleMessageListener;
+import uk.gov.ons.tools.rabbit.SimpleMessageSender;
+
+/** Integration tests for action endpoints */
+@Slf4j
+@ContextConfiguration
+@ActiveProfiles("test")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class ActionEndpointIT {
+
+  @Autowired private ResourceLoader resourceLoader;
+
+  @LocalServerPort private int port;
+
+  @Autowired private ObjectMapper mapper;
+
+  @Autowired private AppConfig appConfig;
+
+  @Rule public WireMockRule wireMockRule = new WireMockRule(options().port(18002));
+
+  @ClassRule public static final SpringClassRule SPRING_CLASS_RULE = new SpringClassRule();
+
+  @Rule public final SpringMethodRule springMethodRule = new SpringMethodRule();
+
+  private Mapzer mapzer;
+
+  @Before
+  public void setup() {
+    mapzer = new Mapzer(resourceLoader);
+    initialiseUnirestObjectMapper();
+  }
+
+  @Test
+  public void ensureSampleIDExposed() throws Exception {
+    UUID collexId = UUID.fromString("3116a1bd-3a84-4761-ae30-4916c4e7120a");
```

Comment:
This test is HUGEEEE can it be broken down into smaller methods to become more readable

Reply:
Done